java/java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertToNestedRecordPattern.java

```diff
@@ -0,0 +1,261 @@
+            return recordComponentMap;
+        } else {
+            DeconstructionPatternTree bTree = (DeconstructionPatternTree) pTree;
+            for (PatternTree p : bTree.getNestedPatterns()) {
+                findNested(p, recordComponentMap);
+            }
+        }
+        return recordComponentMap;
+    }
+
+    private static class UserVariables {
+
+        String methodName;
+        String variable;
+
+        UserVariables(String variable, String methodName) {
+            this.variable = variable;
+            this.methodName = methodName;
+        }
+
+        public String getMethodName() {
+            return methodName;
+        }
+
+        public String getVariable() {
+            return variable;
+        }
+    }
+
+    private static final class FixImpl extends JavaFix {
+
+        private final Map<String, List<UserVariables>> userVars;
+        private final Set<TreePathHandle> replaceOccurrences;
+        List<String> localVarList;
+
+        public FixImpl(CompilationInfo info, TreePath main, Set<TreePath> replaceOccurrences, List<String> localVarList, Map<String, List<UserVariables>> userVars) {
+            super(info, main);
+            this.replaceOccurrences = replaceOccurrences.stream().map(tp -> TreePathHandle.create(tp, info)).collect(Collectors.toSet());
+            this.userVars = userVars;
+            this.localVarList = localVarList;
+        }
+
+        @Override
+        protected String getText() {
+            return Bundle.ERR_ConvertToNestedRecordPattern();
+        }
+
+        @Override
+        protected void performRewrite(JavaFix.TransformationContext ctx) {
+            WorkingCopy wc = ctx.getWorkingCopy();
+            TreePath t = ctx.getPath();
+            TypeElement type = null;
+            Map<PatternTree, List<PatternTree>> recordComponentMap = new LinkedHashMap<>();
+            DeconstructionPatternTree recordPattern = (DeconstructionPatternTree) t.getLeaf();
+            recordComponentMap = findNested(recordPattern, recordComponentMap);
+            
+            Set<String> localVars = new HashSet<>(localVarList);
+            for (PatternTree p : recordComponentMap.keySet()) {
+                List<PatternTree> bindTree = new ArrayList<>();
+                BindingPatternTree bTree = (BindingPatternTree) p;
+                VariableTree v = bTree.getVariable();
+                type = (TypeElement) wc.getTrees().getElement(TreePath.getPath(t, v.getType()));
+                if (type == null || type.getRecordComponents().size() == 0) {
+                    continue;
+                }
+                outer:
+                for (RecordComponentElement recordComponent : type.getRecordComponents()) {
+                    String name = recordComponent.getSimpleName().toString();
+                    String returnType = recordComponent.getAccessor().getReturnType().toString();
+                    returnType = returnType.substring(returnType.lastIndexOf(".") + 1);
```

Comment:
There shouldn't be a need to do String manipulation over the (String representation of the) type. Simply keep the `TypeMirror`, and then use `TreeMaker.Type(TypeMirror)`. That method should take care about printing the correct type, with correct imports, etc.

Reply:
addressed